### DISCUSSION TITLE
Darktheme

### DISCUSSION THREAD

**Discussion by luanayasmim** [luanayasmim/portfolio-aline#4] [Ideas]
How do I do a darktheme with Vue.js and tailwind.css?

**Comment by LuisFernandoPBPereira** [comment]
Is there a lib that do it?

**Reply by luanayasmim** [reply]
Maybe

**Comment by tayllana** [comment]
Creating a dark theme with Vue.js and Tailwind CSS involves setting up a toggleable mechanism to switch between light and dark modes, and then applying appropriate styles to elements based on the selected mode. Here's a step-by-step guide to achieve this:

1. **Install Tailwind CSS**: If you haven't already, install Tailwind CSS in your Vue.js project. You can do this via npm or yarn:
   ```bash
   npm install tailwindcss
   ```
   or
   ```bash
   yarn add tailwindcss
   ```

2. **Set up Tailwind Configuration**: Follow the official Tailwind CSS documentation to set up Tailwind CSS in your Vue.js project. This typically involves creating a `tailwind.config.js` file and configuring Tailwind according to your project requirements.

3. **Create Dark Theme Styles**: Define dark theme styles in your Tailwind CSS configuration or in a separate CSS file. You can specify dark mode styles using Tailwind's `dark` variant feature. For example:
   ```css
   @tailwind base;
   @tailwind components;
   
   @tailwind utilities;
   
   @layer utilities {
     .dark-mode {
       @apply dark:bg-gray-800 dark:text-white;
     }
   }
   ```

4. **Implement Dark Mode Toggle**: Create a mechanism to toggle between light and dark modes. This could be a button or a toggle switch. You can use Vue.js to handle the toggle logic:
   ```html
   <template>
     <button @click="toggleDarkMode">Toggle Dark Mode</button>
     <div :class="{ 'dark-mode': darkMode }">
       <!-- Your content goes here -->
     </div>
   </template>
   
   <script>
   export default {
     data() {
       return {
         darkMode: false
       };
     },
     methods: {
       toggleDarkMode() {
         this.darkMode = !this.darkMode;
       }
     }
   };
   </script>
   ```

5. **Apply Dark Mode Classes Dynamically**: Use Vue.js to dynamically apply dark mode classes to elements based on the `darkMode` data property:
   ```html
   <template>
     <div :class="{ 'dark-mode': darkMode }">
       <button @click="toggleDarkMode">Toggle Dark Mode</button>
       <div :class="{ 'dark:text-white': darkMode }">
         <!-- Your content goes here -->
       </div>
     </div>
   </template>
   ```

6. **Test and Customize**: Test your dark theme implementation thoroughly, and customize styles and toggle mechanisms as needed to match your design requirements.

By following these steps, you can implement a dark theme in your Vue.js project using Tailwind CSS, allowing users to switch between light and dark modes dynamically.

**Reply by luanayasmim** [reply]
Thanks, I did the darktheme works very easily
